streams/src/main/java/org/apache/kafka/streams/state/internals/AbstractSegments.java

```diff
@@ -152,6 +153,13 @@ public List<S> allSegments(final boolean forward) {
         return result;
     }
 
+    @Override
+    public long approximateNumUncommittedBytes() {
```

Comment:
I've added some tests, but since it's not possible to mock the segments themselves, and by default they just return `0` uncommitted bytes, it's not really possible to test that it's correctly summing this until it's possible to enable transaction buffers in the future. I've added a note that the test will consequently need updating later.